### PR TITLE
New post: 2024-04-02-cn9130-temp-regulation

### PR DESCRIPTION
**Pull request by wkz**
Summarize results of a mini study of the default temperature regulation behavior on CN9130.

### CODE REVIEW

File: content/posts/2024-04-02-cn9130-temp-regulation.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+```
+
+This defines the policy of how the kernel's [thermal][3] subsystem is
+allowed to throttle the CPU frequency of a core in response to high
+temperature environments. First, two trip points are defined:
+
+- `cpu0-hot` at 85°C
+- `cpu0-emerg` at 95°C
+
+These are serve as triggers for enforcing different CPU frequency
```
Comment (jovatn) [inline content/posts/2024-04-02-cn9130-temp-regulation.md]: These are triggers? These serve as triggers?

File: content/posts/2024-04-02-cn9130-temp-regulation.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+limitations via the set of defined `cooling-maps`. Deciphering this,
+we see that:
+
+- When `cpu0-hot` trips, i.e. core 0's temperature reaches 85°C, then
+  we enable two `cooling-device`s (`cpu0` and `cpu1`). At this point
```
Comment (jovatn) [inline content/posts/2024-04-02-cn9130-temp-regulation.md]: `cooling devices`

**Comment by jovatn** [comment]
This is really great! 
I did not understand when frequency 533 MHz is applied, but that is not important and likely due to me rather than the text. So go ahead and merge! Perhaps fixing possible typos I spotted and commented on above.

File: content/posts/2024-04-02-cn9130-temp-regulation.md (hunk 2 above)
Comment (wkz) [reply content/posts/2024-04-02-cn9130-temp-regulation.md]: I was going for a plural of the property name in the device tree, hence the trailing "s" outside of the verbatim portion. But I think it just complicates things. Thanks!

**Comment by wkz** [comment]
> This is really great! I did not understand when frequency 533 MHz is applied, but that is not important and likely due to me rather than the text. So go ahead and merge! Perhaps fixing possible typos I spotted and commented on above.

My understanding is that the kernel is free to choose either of the two. So I think it might use it when ramping up the frequency after a period of having it throttled, for example. To limit the risk of overshoots. Power-saving governors other than `performance` (which is the default) might also make use of it, maybe :man_shrugging:

**Pull request merged**